lib/run_loop/device.rb

```diff
@@ -393,6 +402,89 @@ def simulator_wait_for_stable_state
       end
     end
 
+    # @!visibility private
+    #
+    # Sets the AppleLocale key in the .GlobalPreferences.plist file
+    #
+    # @param [String] locale_code a locale code
+    #
+    # @return [RunLoop::Locale] a locale instance
+    #
+    # @raise [RuntimeError] if this is a physical device
+    # @raise [ArgumentError] if the locale code is invalid
+    def simulator_set_locale(locale_code)
+      if physical_device?
+        raise RuntimeError, "This method is for Simulators only"
+      end
+
+      locale = RunLoop::Locale.locale_for_code(locale_code, self)
+
+      global_plist = simulator_global_preferences_path
+      pbuddy.plist_set("AppleLocale", "string", locale.code, global_plist)
+
+      locale
+    end
+
+    # @!visibility private
+    #
+    # Returns the AppleLanguages array in global plist as an array
+    #
+    # @return [Array<String>] list of language codes
+    def simulator_languages
+      global_plist = simulator_global_preferences_path
+      out = pbuddy.plist_read("AppleLanguages", global_plist)
+
+      # example: "Array {\n    en\n    en-US\n}"
+      # I am intentionally punting on this because I don't want
+      # to track down edge cases until the output of this method
+      # is actually used.
+      result = [out]
```

Comment:
Using rescue to catch arbitrary parsing errors is not cool, but as I mention in the comment we are not using this value other than to report to the user the languages.